# A unified approach to evolutionary conservation and population constraint in proteins
Repository containing notebooks to compute statistics in the paper "A unified approach to evolutionary conservation and population constraint in proteins".

## Dataset
The analysis is based on aggregated statistics we computed from data accessed from the following databases:
- [Pfam-A](https://pfam.xfam.org/) database of protein families (version 31.0)
- [gnomAD](https://gnomad.broadinstitute.org/) database of human genetic variation (version 2.1.1).
- [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/) database of human genetic variants and their clinical significance.
- [PDBe](https://www.ebi.ac.uk/pdbe/) database of protein structures.

These were processed into a single dataset of aggregated statistics for each Pfam domain, which is provided in `data/pfam-gnomAD-clinvar-pdb-colstats_c7c3e19.csv.gz`.

## Manuscript figures

The figures in the manuscript are generated by the notebooks in the figure folders under [manuscript-figures](manuscript-figures/).

- [Figure 1B](manuscript-figures/fig1/Fig1B_pfam-residue-missense-hist.md): Frequency distribution of gnomAD missense variants across all amino acid residues in Pfam domains.
- [Figure 1C](manuscript-figures/fig1/Figs1C-D_variant-totals-in-MSA-columns.md): Frequency distributions of gnomAD missense variants over alignment columns of Pfam domains.
- [Figure 1D](manuscript-figures/fig1/Figs1C-D_variant-totals-in-MSA-columns.md): Total number of gnomAD missense or synonymous variants vs. the Shenkin diversity at each position across SH2 domains.
- [Figure 2A](manuscript-figures/fig2/Fig2A_RSA-Class-Distribution.md): Cumulative distributions of the normalised missense enrichment score or normalised Shenkin for positions where the consensus relative solvent accessibility class is core, partially exposed, or surface.
- [Figure 3A](manuscript-figures/fig3/Fig3A_Conservation-Plane.md): The conservation plane: classifying residues in Pfam domains with evolutionary conservation and population constraint.
- [Figure 4A](manuscript-figures/fig4/Fig4A_BioLiP-XClass-Enrichments.md): Odds ratios of the enrichment of protein-ligand interacting residues from BioLiP within sites in different conservation plane categories.
- [Figure 4B](manuscript-figures/fig4/Fig4B_PPI-XClass-Enrichments.md): PPI site enrichments.
- [Figure 4C](manuscript-figures/fig4/Fig4C_ClinVar-XClass-Enrichments.md): ClinVar Pathogenic site enrichments relative to the gnomAD missense background.

## License
This repository and its contents were created by Stuart A. MacGowan ([@stuartmac](https://www.github.com/stuartmac)) at the University of Dundee and is provided under the MIT license. See [LICENSE](LICENSE) for details.